src/strings.rs

```diff
@@ -0,0 +1,122 @@
+use std::ops::Deref;
+use std::borrow::{ToOwned, Borrow};
+use std::str::from_utf8_unchecked;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct StackStr {
+    inner: [u8; 4],
+    len: usize,
+}
+
+impl From<char> for StackStr {
+    fn from(c: char) -> Self {
+        let mut inner = [0u8; 4];
+        c.encode_utf8(&mut inner);
+        Self {
+            inner,
+            len: c.len_utf8(),
+        }
+    }
+}
+
+impl Deref for StackStr {
+    type Target = str;
+
+    fn deref(&self) -> &str {
+        unsafe {
+            from_utf8_unchecked(&self.inner[..self.len])
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
```

Comment:
The name is whimsical, but I think confusing, I wonder if we can come up with a better one.

Reply:
Yes! Naming things is hard. Very open to any other name :-)

Reply:
"Smort" is a bit weird.